src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteSpec.groovy

```diff
@@ -1,185 +1,179 @@
-import static org.openkilda.functionaltests.helpers.Wrappers.retry
-import static org.openkilda.functionaltests.helpers.Wrappers.timedLoop
-import static org.openkilda.functionaltests.helpers.Wrappers.wait
-import static org.openkilda.messaging.info.event.IslChangeType.FAILED
-import static org.openkilda.testing.Constants.PATH_INSTALLATION_TIME
-import static org.openkilda.testing.Constants.WAIT_OFFSET
-import static org.openkilda.testing.service.floodlight.model.FloodlightConnectMode.RW
-
 @Slf4j
 @Narrative("Verify different cases when Kilda is supposed to automatically reroute certain flow(s).")
 
 class AutoRerouteSpec extends HealthCheckSpecification {
 
+    @Autowired
+    @Shared
+    FlowFactory flowFactory
+
     @Tags([SMOKE, ISL_RECOVER_ON_FAIL])
     @IterationTag(tags = [TOPOLOGY_DEPENDENT], iterationNameRegex = /vxlan/)
     def "Flow is rerouted when one of the #description flow ISLs fails"() {
         given: "A flow with one alternative path at least"
-        def data = flowData(switchPairs.all().neighbouring(), 1)
-        FlowRequestV2 flow = data[0]
-        def allFlowPaths = data[1]
-        flowHelperV2.addFlow(flow)
-        def flowPath = PathHelper.convert(northbound.getFlowPath(flow.flowId))
+        FlowExtended flow = flowFactory.getRandom(swicthPair)
+        def allFlowPaths = swicthPair.getPaths()
+        def initialPath = flow.retrieveAllEntityPaths()
 
         when: "Fail a flow ISL (bring switch port down)"
         Set<Isl> altFlowIsls = []
-        def flowIsls = pathHelper.getInvolvedIsls(flowPath)
-        allFlowPaths.findAll { it != flowPath }.each { altFlowIsls.addAll(pathHelper.getInvolvedIsls(it)) }
+        def flowIsls = initialPath.flowPath.getInvolvedIsls()
+        allFlowPaths.findAll { it != initialPath.getPathNodes() }
+                .each { altFlowIsls.addAll(pathHelper.getInvolvedIsls(it)) }
         def islToFail = flowIsls.find { !(it in altFlowIsls) && !(it.reversed in altFlowIsls) }
         islHelper.breakIsl(islToFail)
 
         then: "The flow was rerouted after reroute delay"
         wait(rerouteDelay + WAIT_OFFSET) {
-            assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.UP
-            assert PathHelper.convert(northbound.getFlowPath(flow.flowId)) != flowPath
+            assert flow.retrieveFlowStatus().status == FlowState.UP
+            assert flow.retrieveAllEntityPaths() != initialPath
         }
 
         where:
-        description | flowData
-        "vlan"      | { SwitchPairs switchPairs, Integer minAltPathsCount ->
-            getFlowWithPaths(switchPairs, minAltPathsCount)
-        }
-        "vxlan"     | { SwitchPairs switchPairs, Integer minAltPathsCount ->
-            getVxlanFlowWithPaths(switchPairs, minAltPathsCount)
-        }
+        description | swicthPair
+        "vlan"      | switchPairs.all().neighbouring().withAtLeastNPaths(2).random()
```

Comment:
withAtLeastNPaths(**2**) 
1 from  flowData(switchPairs.all().neighbouring(), **1**) from the test (deleted line 56)
+1 from getFlowWithPaths(switchPairs, minAltPathsCount) (deleted line 590)